src/odb/src/db/dbSite.cpp

```diff
@@ -230,6 +232,50 @@ void dbSite::setClass(dbSiteClass type)
   site->_flags._class = type.getValue();
 }
 
+void dbSite::setRowPattern(
+    std::vector<std::pair<std::string, std::string>>& row_pattern)
```

Comment:
I think it would be better to replace std::pair with a class named something like RowPatternElement where the definition is something like.
```c++
class RowPatternElement {
public:
  RowPatternElement(dbSite*, dbOrientType)
}
```

I think it would be a good idea to force the user to provide a dbSite* and dbOrientType upfront. That way ODB doesn't need to do as much error handling. As it is you can set an invalid name, and it will just get silently removed during get.

Reply:
I agree, that is much easier to understand

Reply:
The conversion to RowPatternElement  should happen in lefin along with validation